### PR TITLE
Generic dungeons mechanism, OX Freighter

### PR DESCRIPTION
**Pull request by DreamySkrell**
image
image

image
image

# Explanation:
In the OX freighter, the containers are randomized, and have different contents and layouts every round. The containers are picked from a list.

As for how the mechanism itself works, the landmarks add themselves to a global landmark list at `New()`. At `Initialize()`, the dungeons are actually loaded into the maps.

The mechanism is flexible enough. Spawn chance can be configured, the dungeon map files can be specified one by one or a whole directory can be provided. And dungeons/landmarks can be marked as "unique", meaning that a dungeon is blacklisted after it spawns and cannot be spawned again by other landmark.

# Checklist:
- [x] first impl
- [x] make sure this is actually wanted
- [x] check and note how does it work actually (what things does it load/copy, like areas) - noted in comments in code
- [x] make sure it can't be misused - mappers only subtype the landmark, and do not call any functions or anything
- [x] test spawning in multiple dungeons at once - the freighter spawns 8 dungeons, so everything should be fine I think
- [x] create a container ship that showcases the functionality
- [ ] maybe test merge to see the performance impact

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!wip

**Comment by DreamySkrell** [comment]
!review

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/controllers/subsystems/initialization/map_dungeons_generic.dm [hunk 1]
```diff
@@ -0,0 +1,110 @@
+// -------------------------------------------------------------------
+
+//
+var/list/dungeon_generic_landmarks = list()
+var/list/dungeon_generic_initialized = FALSE
```
[the remark this answers is not in the document]
Comment (DreamySkrell) [reply code/controllers/subsystems/initialization/map_dungeons_generic.dm:5]: These are used in
`/obj/effect/dungeon_generic_landmark/Initialize()`
and
```
proc/place_dungeons_generic()
	if(dungeon_generic_initialized)
		return
    ...
```

**Comment by JohnWildkins** [comment]
See: Discussion in discord about current code / making this a subsystem

**Comment by NonQueueingMatt** [comment]
Inactive

**Pull request closed without merging**